src/figuresClasses.ts

```diff
@@ -1,11 +1,83 @@
-export interface Figure {}
+type Shapes = 'triangle' | 'circle' | 'rectangle';
+type Colors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: Shapes;
+  color: Colors;
+  getArea(): number;
+}
+
+function checkError(condition: boolean, message: string): void {
+  if (condition) {
+    throw new Error(message);
+  }
+}
+
+export class Triangle implements Figure {
+  readonly shape = 'triangle';
+
+  constructor(
+    public color: Colors,
+    private a: number,
+    private b: number,
+    private c: number,
+  ) {
+    checkError(a <= 0 || b <= 0 || c <= 0, 'One or more length <= 0');
```

Comment:
Better to make error messages clearer and more informative:

```suggestion
    checkError(a <= 0 || b <= 0 || c <= 0, 'One or more sides are less than or equal to zero');
```